src/main/java/org/inferred/freebuilder/processor/DefaultPropertyFactory.java

```diff
@@ -169,18 +169,56 @@ public void addFinalFieldAssignment(SourceBuilder code, String finalField, Strin
 
     @Override
     public void addMergeFromValue(Block code, String value) {
-      code.addLine("%s(%s.%s());", setter(property), value, property.getGetterName());
+      Excerpt defaults = Declarations.freshBuilder(code, metadata).orNull();
+      if (defaults != null) {
+        code.add("if (");
+        if (!hasDefault) {
+          code.add("%s._unsetProperties.contains(%s.%s) || ",
```

Comment:
The value of `builder.a` is undefined if `builder._unsetProperties.contains(A)`. For instance, for primitive types. there is no "null" to default to.

`b.getA().isPresent()` should be `true` as the unset property on `a` will be ignored when merging.